File: server/datasets/java_source-code/Multiple-G3/renamed_42_dsu.java
public class DSU_6 {
    public static void main(String[] args) {
        DSU_6 dsu = new DSU_6(10);
        dsu.u_6(1, 2);
    }
    int[] p_6;
    int[] r_6;
    public DSU_6(int tot_6) {
        p_6 = new int[tot_6];
        r_6 = new int[tot_6];
        for (int idx = 0; idx < tot_6; idx++) {
            p_6[idx] = idx;
            r_6[idx] = 0;
        }
    }
    public int f_6(int x_6) {
        if (p_6[x_6] != x_6) {
            p_6[x_6] = f_6(p_6[x_6]);
        }
        return p_6[x_6];
    }
    public void u_6(int a_6, int b_6) {
        int ra_6 = f_6(a_6);
        int rb_6 = f_6(b_6);
        if (ra_6 != rb_6) {
            if (r_6[ra_6] < r_6[rb_6]) {
                p_6[ra_6] = rb_6;
            } else if (r_6[ra_6] > r_6[rb_6]) {
                p_6[rb_6] = ra_6;
            } else {
                p_6[rb_6] = ra_6;
                r_6[ra_6]++;
            }
        }
    }
}

File: server/datasets/controlled_benchmarks/python/multiple_files/multiple_g3/renamed_42_dsu.py
class DSU_6:
    def __init__(self, n_6):
        self.p_6 = list(range(n_6))
        self.rk_6 = [0] * n_6

    def find_6(self, x_6):
        if self.p_6[x_6] != x_6:
            self.p_6[x_6] = self.find_6(self.p_6[x_6])
        return self.p_6[x_6]

    def union_6(self, x_6_a, x_6_b):
        ra_6 = self.find_6(x_6_a)
        rb_6 = self.find_6(x_6_b)
        if ra_6 != rb_6:
            if self.rk_6[ra_6] < self.rk_6[rb_6]:
                ra_6, rb_6 = rb_6, ra_6
            self.p_6[rb_6] = ra_6
            if self.rk_6[ra_6] == self.rk_6[rb_6]:
                self.rk_6[ra_6] += 1
            return True
        return False
